src/sentry/static/sentry/app/views/organizationEventsV2/events.jsx

```diff
@@ -23,15 +23,61 @@ const CHART_AXIS_OPTIONS = [
   {label: 'Users', value: 'user_count'},
 ];
 
-export default class Events extends AsyncComponent {
+export default class Events extends React.Component {
   static propTypes = {
     router: PropTypes.object.isRequired,
     location: PropTypes.object.isRequired,
     organization: SentryTypes.Organization.isRequired,
     view: SentryTypes.EventView.isRequired,
   };
 
-  shouldReload = true;
+  render() {
+    const {organization, view, location, router} = this.props;
+    const query = location.query.query || '';
+
+    return (
+      <React.Fragment>
+        <Panel>
+          {getDynamicText({
+            value: (
+              <EventsChart
+                router={router}
+                query={query}
+                organization={organization}
+                showLegend
+                yAxisOptions={CHART_AXIS_OPTIONS}
+              />
+            ),
+            fixed: 'events chart',
+          })}
+        </Panel>
+        <StyledSearchBar
+          organization={organization}
+          query={query}
+          onSearch={this.handleSearch}
+        />
+        <Container>
+          <EventsTable
+            key={view.id}
+            location={location}
+            organization={organization}
+            view={view}
+          />
+          <Tags view={view} organization={organization} location={location} />
+        </Container>
+      </React.Fragment>
+    );
+  }
+}
+
+class EventsTable extends AsyncComponent {
```

Comment:
@markstory I've added more changes to this PR to address more than just a proptype warning.

When a user switches between events v2 tabs (e.g. "All Events" to "Errors"), the `view` prop changes before the `data` state provided by the `AsyncComponent` is properly updated. This causes a render to occur where the `view` prop is incorrectly paired with the `data` state.

To address this, I've isolated the `AsyncComponent` component and shrunk its scope to just the components that would consume its state and props. 

Due to the nature of the `` component, I'm remounting `<EventsTable />` by using the key, e.g. `key={view.id}`.